### PR TITLE
Document roziscoding/grammy-autoquote

### PR DESCRIPTION
**Pull request by roziscoding**
I used google translate for spanish and chinese translations as I don't speak anny of those languages, so it's okay if they need to be replaced by actual translations

### CODE REVIEW

**Comment by KnorpelSenf** [comment]
It is not necessary to provide machine translations. We would like to have accurate translations written by native or near-native speakers. Feel free to remove these files again, they serve no purpose.

Other contributors will take care of the translations as soon as the English version is approved by the website maintainer.

**Comment by roziscoding** [comment]
Okay. Should I also remove the entries for the other languages from the navbar?

**Comment by roziscoding** [comment]
Okay. Should I also remove the entries for the other languages from the navbar?

**Comment by KnorpelSenf** [comment]
You can leave empty entries there for the translators to fill out if you want

**Comment by KnorpelSenf** [comment]
Please run the formatting script according to the README to fix CI :)
https://github.com/grammyjs/website#checking-correct-formatting-of-docs

**Comment by roziscoding** [comment]
I did 🤔 
But I'll do it again, anyways

**Comment by roziscoding** [comment]
image

Does `deno fmt` only check changed files or something? Or maybe is the deno version on the CI different from my version (1.24.0)?

**Comment by roj1512** [comment]
@roziscoding yep, you should upgrade:

```shell
deno upgrade
```

**Comment by roziscoding** [comment]
I'm already on 1.24.0 (which `deno upgrade` says is the latest version)

**Review by KnorpelSenf — APPROVED** [review]

File: site/docs/plugins/autoquote.md
Comment (roj1512) [inline site/docs/plugins/autoquote.md:3]:
```suggestion
Sometimes, especially for bots that are meant to be used in groups, it is necessary to always send messages as a reply (or quote) to the message that started the interaction.
Usually, the way to do that is to mannualy add `reply_to_message_id` to the params of the mathod that sends the message (`sendText` / `reply`, `sendPhoto` / `replyWithPhoto`).
However, if you're doing this for every message, it can get a bit messy and tiring.
```

File: site/docs/plugins/autoquote.md
Comment (roj1512) [inline site/docs/plugins/autoquote.md:1]: Forcing users to reply (`autoquote`)

Comment (roziscoding) [reply site/docs/plugins/autoquote.md:1]: It's the bot that it forced to reply, though. The next sent messages are going to quote the message that triggered the update

Comment (roj1512) [reply site/docs/plugins/autoquote.md:1]: brb. will look into it tomorrow.

File: site/docs/plugins/autoquote.md
Comment (CikiMomogi) [inline site/docs/plugins/autoquote.md:4]: Typo `mathod`

File: site/docs/plugins/autoquote.md
Comment (CikiMomogi) [inline site/docs/plugins/autoquote.md:18]: Change single quote to double quote (https://github.com/grammyjs/website/pull/389)

**Review by CikiMomogi — CHANGES_REQUESTED** [review]

File: site/docs/plugins/autoquote.md
Comment (CikiMomogi) [reply site/docs/plugins/autoquote.md:1]: Always Replying to Messages

File: site/docs/plugins/autoquote.md
Comment (CikiMomogi) [inline site/docs/plugins/autoquote.md:11]: For a Single Route

**Review by CikiMomogi — CHANGES_REQUESTED** [review]
Title case

File: site/docs/plugins/autoquote.md
Comment (roziscoding) [reply site/docs/plugins/autoquote.md:4]:
```suggestion
Usually, the way to do that is to mannualy add `reply_to_message_id` to the params of the method that sends the message (`sendText` / `reply`, `sendPhoto` / `replyWithPhoto`).
```

File: site/docs/plugins/autoquote.md
Comment (roziscoding) [reply site/docs/plugins/autoquote.md:18]:
```suggestion
// import { addReplyParam } from "https://deno.land/x/grammy_autoquote/mod.ts"
```

File: site/docs/plugins/autoquote.md
Comment (roziscoding) [reply site/docs/plugins/autoquote.md:11]:
```suggestion
### For a Single Route
```

File: site/docs/plugins/autoquote.md
Comment (roziscoding) [reply site/docs/plugins/autoquote.md:1]:
```suggestion
# Always Replying to Messages
```

File: site/docs/plugins/autoquote.md
Comment (CikiMomogi) [inline site/docs/plugins/autoquote.md:30]: Oops.. I missed this one. Title case :)

Comment (roziscoding) [reply site/docs/plugins/autoquote.md:30]:
```suggestion
### Usage for Every Route
```
Isn't it possible to do this vis css tho?

File: site/docs/plugins/autoquote.md
Comment (KnorpelSenf) [inline site/docs/plugins/autoquote.md:24]:
```suggestion
  await ctx.reply("Demo command!"); // This will quote the user's message
```

File: site/docs/plugins/autoquote.md
Comment (KnorpelSenf) [inline site/docs/plugins/autoquote.md:44]:
```suggestion
  await ctx.reply("Demo command!"); // This will quote the user's message
```

File: site/docs/plugins/autoquote.md
Comment (KnorpelSenf) [inline site/docs/plugins/autoquote.md:48]:
```suggestion
  await ctx.reply("Hi there :)"); // Also quotes the user's message
```

**Review by KnorpelSenf — CHANGES_REQUESTED** [review]
I just noticed I have a few more nitpicks, but otherwise this LGTM.

@roj1512 should the third-party modules be sorted chronologically or alphabetically?

**Comment by roj1512** [comment]
@KnorpelSenf We decide on that later

**Comment by KnorpelSenf** [comment]
In other words, chronologically until we decide to reorder?

**Comment by roj1512** [comment]
Yes, until we start deciding to reorder after merging this

**Comment by KnorpelSenf** [comment]
Does that mean this is ready for translation?

**Comment by roj1512** [comment]
No, not without your suggestions being applied.

**Comment by KnorpelSenf** [comment]
@roziscoding I took the liberty to fix up the examples

**Comment by roziscoding** [comment]
> @roziscoding I took the liberty to fix up the examples

I'm sorry, I hadn't seen these. Thanks for making the changes tho

**Review by KnorpelSenf — CHANGES_REQUESTED** [review]
Please add JS and Deno examples wherever the code needs to be runtime-specific. Otherwise, this LGTM.

**Comment by roziscoding** [comment]
> Please add JS and Deno examples wherever the code needs to be runtime-specific. Otherwise, this LGTM.

Done!

**Review by KnorpelSenf — APPROVED** [review]

File: site/docs/plugins/autoquote.md
Comment (roj1512) [inline site/docs/plugins/autoquote.md:3]:
```suggestion
It is sometimes necessary to always send messages as replies, especially for bots that are meant to be used in groups.
```

File: site/docs/plugins/autoquote.md
Comment (roj1512) [inline site/docs/plugins/autoquote.md:4]:
```suggestion
We usually do this by adding the `reply_to_message_id` parameter to the methods that send the message: `sendText`, `reply`, sendPhoto, replyWithPhoto and etc.
```

File: site/docs/plugins/autoquote.md
Comment (roj1512) [inline site/docs/plugins/autoquote.md:5]:
```suggestion
However, if you're doing this for every single message, it can get messy and boring.
```

File: site/docs/plugins/autoquote.md
Comment (roj1512) [inline site/docs/plugins/autoquote.md:7]:
```suggestion
This plugin sets the `reply_to_message_id` parameter to `ctx.msg.message_id` for `reply` and all `send*` methods that support it to make every message a reply to the message that triggered it.
```

File: site/docs/plugins/autoquote.md
Comment (roj1512) [inline site/docs/plugins/autoquote.md:11]:
```suggestion
### In a Single Route
```

File: site/docs/plugins/autoquote.md
Comment (roj1512) [inline site/docs/plugins/autoquote.md:13]:
```suggestion
If you want all messages sent within a specific context (like a specific command), you can apply it to them specifically: 
```

File: site/docs/plugins/autoquote.md
Comment (roj1512) [inline site/docs/plugins/autoquote.md:69]:
```suggestion
### In for All Routes
```

File: site/docs/plugins/autoquote.md
Comment (roj1512) [inline site/docs/plugins/autoquote.md:71]:
```suggestion
If you want every single message that is sent reply the messages that triggered them, you can apply the plugin this way:
```

File: site/docs/plugins/autoquote.md
Comment (roj1512) [inline site/docs/plugins/autoquote.md:133]:
```suggestion
      await ctx.reply("Hi there :)"); // this quotes the user's message, too
```

File: site/docs/plugins/autoquote.md
Comment (roj1512) [inline site/docs/plugins/autoquote.md:129]:
```suggestion
      await ctx.reply("Demo command!"); // this is going to quote the user's message
```

File: site/docs/plugins/autoquote.md
Comment (roj1512) [inline site/docs/plugins/autoquote.md:111]:
```suggestion
      await ctx.reply("Hi there :)"); // this quotes the user's message, too
```

File: site/docs/plugins/autoquote.md
Comment (roj1512) [inline site/docs/plugins/autoquote.md:107]:
```suggestion
      await ctx.reply("Demo command!"); // this is going to quote the user's message
```

File: site/docs/plugins/autoquote.md
Comment (roj1512) [inline site/docs/plugins/autoquote.md:85]:
```suggestion
      await ctx.reply("Demo command!"); // this is going to quote the user's message
```

File: site/docs/plugins/autoquote.md
Comment (roj1512) [inline site/docs/plugins/autoquote.md:89]:
```suggestion
      await ctx.reply("Hi there :)"); // this quotes the user's message, too
```

File: site/docs/plugins/autoquote.md
Comment (roj1512) [inline site/docs/plugins/autoquote.md:60]:
```suggestion
      await ctx.reply("Demo command!"); // this is going to quote the user's message
```

File: site/docs/plugins/autoquote.md
Comment (roj1512) [inline site/docs/plugins/autoquote.md:43]:
```suggestion
      await ctx.reply("Demo command!"); // this is going to quote the user's message
```

File: site/docs/plugins/autoquote.md
Comment (roj1512) [inline site/docs/plugins/autoquote.md:26]:
```suggestion
      await ctx.reply("Demo command!"); // this is going to quote the user's message
```

File: site/docs/plugins/autoquote.md
Comment (roj1512) [inline site/docs/plugins/autoquote.md:11]:
```suggestion
### In Specific Routes
```

**Review by roj1512 — APPROVED** [review]
LGTM. Thank you so much!

**Comment by roj1512** [comment]
f. what just happened

**Comment by roj1512** [comment]
@roziscoding Can you please reopen?

**Comment by roziscoding** [comment]
GitHub won't let me.
I'll create a new PR

**Comment by roj1512** [comment]
Yeah yeah, I meant a new PR.

**Comment by roziscoding** [comment]
I think the review changes are gone... At least the latest ones 🤔 
I can't find them locally after doing `git pull`

**Comment by roziscoding** [comment]
I `git push`ed again so I can reopen, but I don't know what's been lost 😅

**Comment by roziscoding** [comment]
Yeah, should be okay now.
Good thing GitHub keeps resolved comments, so I just reapplied the suggestions :)

**Comment by roziscoding** [comment]
image

**Comment by roj1512** [comment]
AHAHAH. Darlene's brother.

**Comment by roj1512** [comment]
AHAHAH. Darlene's brother.

**Review by roj1512 — APPROVED** [review]

File: site/docs/plugins/autoquote.md
Comment (CikiMomogi) [inline site/docs/plugins/autoquote.md:4]:
```suggestion
We usually do this by adding the `reply_to_message_id` parameter to the methods that send the message: `sendText`, `reply`, `sendPhoto`, `replyWithPhoto` and etc.
```

File: site/docs/plugins/autoquote.md
Comment (CikiMomogi) [inline site/docs/plugins/autoquote.md:7]:
```suggestion
This plugin sets the `reply_to_message_id` parameter to `ctx.msg.message_id` for all `reply*` and `send*` methods that support it to make every message a reply to the message that triggered it.
```

**Review by CikiMomogi — CHANGES_REQUESTED** [review]

**Comment by CikiMomogi** [comment]
Hmm.. orange?! 

21312dsads

File: site/docs/plugins/autoquote.md
Comment (CikiMomogi) [inline site/docs/plugins/autoquote.md:30]:
```suggestion
```ts
import { Bot } from "grammy";
import { addReplyParam } from "@roziscoding/grammy-autoquote";

const bot = new Bot("");

bot.command("demo", async (ctx) => {
  ctx.api.config.use(addReplyParam(ctx));
  await ctx.reply("Demo command!"); // this is going to quote the user's message
});

bot.start();
```
```

File: site/docs/plugins/autoquote.md
Comment (CikiMomogi) [inline site/docs/plugins/autoquote.md:47]:
````suggestion
```js
const { Bot } = require("grammy");
const { addReplyParam } = require("@roziscoding/grammy-autoquote");

const bot = new Bot("");

bot.command("demo", async (ctx) => {
  ctx.api.config.use(addReplyParam(ctx));
  await ctx.reply("Demo command!"); // this is going to quote the user's message
});

bot.start();
```
````

File: site/docs/plugins/autoquote.md
Comment (CikiMomogi) [inline site/docs/plugins/autoquote.md:137]:
````suggestion
```ts
import { Bot } from "https://deno.land/x/grammy/mod.ts";
import { autoQuote } from "https://deno.land/x/grammy_autoquote/mod.ts";

const bot = new Bot("");

bot.use(autoQuote);

bot.command("demo", async (ctx) => {
  await ctx.reply("Demo command!"); // this is going to quote the user's message
});

bot.command("hello", async (ctx) => {
  await ctx.reply("Hi there :)"); // this quotes the user's message, too
});

bot.start();
```
````

**Review by CikiMomogi — CHANGES_REQUESTED** [review]
Fix the orange background in the code snippets.

File: site/docs/plugins/autoquote.md
Comment (CikiMomogi) [inline site/docs/plugins/autoquote.md:115]:
````suggestion
```js
const { Bot } = require("grammy");
const { autoQuote } = require("@roziscoding/grammy-autoquote");

const bot = new Bot("");

bot.use(autoQuote);

bot.command("demo", async (ctx) => {
  await ctx.reply("Demo command!"); // this is going to quote the user's message
});

bot.command("hello", async (ctx) => {
  await ctx.reply("Hi there :)"); // this quotes the user's message, too
});

bot.start();
```
````

File: site/docs/plugins/autoquote.md
Comment (CikiMomogi) [inline site/docs/plugins/autoquote.md:93]:
````suggestion
```ts
import { Bot } from "grammy";
import { autoQuote } from "@roziscoding/grammy-autoquote";

const bot = new Bot("");

bot.use(autoQuote);

bot.command("demo", async (ctx) => {
  await ctx.reply("Demo command!"); // this is going to quote the user's message
});

bot.command("hello", async (ctx) => {
  await ctx.reply("Hi there :)"); // this quotes the user's message, too
});

bot.start();
```
````

File: site/docs/plugins/autoquote.md
Comment (CikiMomogi) [inline site/docs/plugins/autoquote.md:64]:
````suggestion
```ts
import { Bot } from "https://deno.land/x/grammy/mod.ts";
import { addReplyParam } from "https://deno.land/x/grammy_autoquote/mod.ts";

const bot = new Bot("");

bot.command("demo", async (ctx) => {
  ctx.api.config.use(addReplyParam(ctx));
  await ctx.reply("Demo command!"); // this is going to quote the user's message
});

bot.start();
```
````

**Review by CikiMomogi — CHANGES_REQUESTED** [review]

**Review by CikiMomogi — CHANGES_REQUESTED** [review]
Add `grammy_autoquote` to [modules.json](https://github.com/grammyjs/website/blob/main/site/docs/.vuepress/plugins/current-versions/modules.json)

**Comment by roj1512** [comment]
Thanks @CikiMomogi.

**Review by roj1512 — APPROVED** [review]

**Comment by CikiMomogi** [comment]
> Add `grammy_autoquote` to [modules.json](https://github.com/grammyjs/website/blob/main/site/docs/.vuepress/plugins/current-versions/modules.json)

@roziscoding

**Comment by roziscoding** [comment]
> > Add `grammy_autoquote` to [modules.json](https://github.com/grammyjs/website/blob/main/site/docs/.vuepress/plugins/current-versions/modules.json?rgh-link-date=2022-08-18T17%3A13%3A06Z)
> 
> @roziscoding

Done!

**Review by CikiMomogi — APPROVED** [review]

File: site/docs/plugins/autoquote.md
Comment (WingLim) [inline site/docs/plugins/autoquote.md:146]:
```suggestion
- API Reference: <https://doc.deno.land/https://deno.land/x/grammy_autoquote/mod.ts>
```

**Review by WingLim — CHANGES_REQUESTED** [review]
we shoud use the same format for plugin summary

**Pull request merged**
